scripts/release/status.js

```diff
@@ -0,0 +1,62 @@
+'use strict'
+
+/* eslint-disable no-console */
+
+const { GITHUB_REF, GITHUB_TOKEN } = process.env
+
+const TIMEOUT = 10 * 1000
+const MAX_ATTEMPTS = 30
+
+let attempts = 0
+
+async function checkStatuses (contexts) {
+  const url = `https://api.github.com/repos/DataDog/dd-trace-js/commits/${GITHUB_REF}/status`
+  const response = await fetch(url, {
+    headers: {
+      Accept: 'application/vnd.github+json',
+      Authorization: `Bearer ${GITHUB_TOKEN}`,
+      'X-GitHub-Api-Version': '2022-11-28'
+    }
+  })
+
+  if (response.status !== 200) {
+    console.log(response)
+    console.log(response.text())
+
+    throw new Error('Could not get status from GitHub.')
+  }
+
+  const { statuses } = JSON.parse(await response.text())
+
+  for (const status of statuses) {
+    for (const context of contexts) {
+      if (status.context === context) {
+        switch (status.state) {
+          case 'success':
+            contexts = contexts.filter(c => c !== context)
+            break
+          case 'cancelled':
+          case 'failure':
+          case 'stale':
+          case 'timed_out':
+            throw new Error(`Job was not successful: ${context}.`)
+        }
+      }
+    }
+  }
+
+  if (contexts.length === 0) return
```

Comment:
An optimistic approach about each status containing a context and the state as such having to succeed without looking it up.

```suggestion
  const contextSet = new Set(contexts)

  for (const status of statuses) {
    const context = contextSet.get(status.context)
    if (status.state !== 'success') {
      throw new Error(`Job was not successful: ${context}.`)
    }
    contextSet.delete(context)
  }
  
  if (contextSet.size === 0) return
```